### PR TITLE
Checked the docs: introduction

### PR DESCRIPTION
**Pull request by banilo**

### CODE REVIEW

**Comment by banilo** [comment]
More generally, there appear to be some hyphenation issues in the docs. Especially, in this and similar scenarios "machine-learning methods"

File: doc/introduction.rst [hunk 1]
```diff
@@ -27,29 +27,32 @@ What is nilearn: MVPA, decoding, predictive models, functional connectivity
...
     For a machine-learning expert, the value of nilearn can be seen as
-    domain-specific **feature engineering** construction, to go from
-    neuroimaging data to a feature-matrix well suited to statistical
+    domain-specific **feature engineering** construction, that is, shaping
+    neuroimaging data into a feature matrix well suited to statistical
     learning, or vice versa.
 
 
-Why is machine learning relevant to NeuroImaging: a few examples
+Why is machine learning relevant to NeuroImaging?: a few examples
```
Comment (lesteve) [inline doc/introduction.rst]: Hmm was `?:` intended, i.e. the question mark followed by a colon ?

File: doc/introduction.rst [hunk 2]
```diff
@@ -75,30 +83,38 @@ Glossary: machine learning vocabulary
...
     
-    * a **regression** problem: predicting a continuous quantity such 
-      as age
+    * a **regression** problem: predicting a continuous variable, such 
+      as participant age, from the data `X`
     
-    * a **classification** problem: predicting the class each 
-      observation belongs to, such as patient or control
+    * a **classification** problem: predicting a binary variable that splits 
+      the observation into two groups, such as patients versus controls
```
Comment (lesteve) [inline doc/introduction.rst]: maybe observation**s** with a final 's'?

File: doc/introduction.rst [hunk 3]
```diff
@@ -75,30 +83,38 @@ Glossary: machine learning vocabulary
...
 
 :Unsupervised learning:
 
     `Unsupervised learning
     <http://scikit-learn.org/stable/unsupervised_learning.html>`_ is
     concerned with data `X` without any labels. It analyzes the structure
-    of a dataset, for instance **clustering** or extracting latent
-    factors such as with **independent components analysis (ICA)**.
+    of a dataset to find coherent underlying structure,
+    for instance using clustering **clustering**, or to extract latent
```
Comment (lesteve) [inline doc/introduction.rst]: clustering repeated twice

File: doc/introduction.rst [hunk 4]
```diff
@@ -107,27 +123,29 @@ Glossary: machine learning vocabulary
 Installing nilearn
 ====================
 
-Installing the Python scientific environment
+Installing the scientific Python environment
```
Comment (lesteve) [inline doc/introduction.rst]: arguable. I don't know exactly why but I preferred it as it was.

File: doc/introduction.rst [hunk 5]
```diff
@@ -158,42 +177,45 @@ a command prompt::
 Testing your installation
 .........................
 
-To test if you have done everything right, open IPython and try the
-following, in the Python prompt::
+To verify whether everything is set up correctly, open IPython (an widely
+used enhanced version of the Python prompt, cf. below) and punch in the follwing
```
Comment (lesteve) [inline doc/introduction.rst]: missing 'o' in foll**o**wing

maybe 'type in' is simpler to understand than 'punch in'

File: doc/introduction.rst [hunk 6]
```diff
@@ -158,42 +177,45 @@ a command prompt::
...
+line::
 
     In [1]: import nilearn
 
-If you do not get any errors, you have installed nilearn correctly.
+If no errors occur, you have installed nilearn correctly.
```
Comment (lesteve) [inline doc/introduction.rst]: google seems to prefer "no error occur**s**"

File: doc/introduction.rst [hunk 7]
```diff
@@ -158,42 +177,45 @@ a command prompt::
...
-As time goes, you can update your copy of nilearn by doing "git pull" in
-this directory.
+In the future, you can readily update your copy of nilearn
+by executing "git pull" in
+the nilearn root directory (yes, as a shell command).
 
-If you really don't want to use git, you download the latest development
-snapshot from the following link and unziping it:
+If you really do not want to use git, you may still download the latest
+developmental snapshot from the following link (unziping required):
```
Comment (lesteve) [inline doc/introduction.rst]: I would stick with development rather than developmental.

File: doc/introduction.rst [hunk 8]
```diff
@@ -266,29 +295,29 @@ the file name::
 
 .. currentmodule:: nilearn
 
-For simple operations on images, there exists many functions, such as in
-the :mod:`nilearn.image` module for image manipulation, eg
+To simple functions/operations on images, many functions exist, such as in
```
Comment (lesteve) [inline doc/introduction.rst]: I would keep "For simple ..." rather than "To simple ..."

File: doc/introduction.rst [hunk 9]
```diff
@@ -266,29 +295,29 @@ the file name::
...
 
     >>> smoothed_img.to_filename("/home/user/t_map000_smoothed.nii")
 
 Finally, nilearn deals with Nifti images that come in two flavors: 3D
-images, that represent a brain volume, and 4D images, that represent a
-series of brain volume. To extract the n-th 3D image in a 4D image, use
-:func:`image.index_img` (keep in mind that indexing starts at 0 in
-Python)::
+images, which represent a brain volume, and 4D images, which represent a
+series of brain volume. To extract the n-th 3D image from a 4D image, you can
```
Comment (lesteve) [inline doc/introduction.rst]: brain volume**s**

File: doc/introduction.rst [hunk 10]
```diff
@@ -266,29 +295,29 @@ the file name::
...
+use the :func:`image.index_img` function (keep in mind that array indexing
+always starts at 0 in the Python language)::
 
     >>> first_volume = image.index_img("/home/user/fmri_volumes.nii", 0)
 
-To do a loop over each volume of a 4D image, use :func:`image.iter_img`::
+To loop over each and every volume of a 4D image, use :func:`image.iter_img`::
```
Comment (lesteve) [inline doc/introduction.rst]: I am not sure what the "and every" brings

File: doc/introduction.rst [hunk 11]
```diff
@@ -298,8 +327,8 @@ To do a loop over each volume of a 4D image, use :func:`image.iter_img`::
 
    Want to sharpen your skills with nilearn? 
    Compute the mean EPI for first subject of the ADHD
-   dataset downloaded with :func:`nilearn.datasets.fetch_adhd`, and
-   smooth it with an FWHM varying from 0mm to 20mm in increments of 5mm
+   dataset downloaded with :func:`nilearn.datasets.fetch_adhd` and
+   smooth it by an FWHM varying from 0mm to 20mm in increments of 5mm
```
Comment (lesteve) [inline doc/introduction.rst]: with -> by replacement arguable

File: doc/introduction.rst [hunk 12]
```diff
@@ -330,15 +359,16 @@ directly to the section you need:
 Scientific computing with Python
 ---------------------------------
 
-If you are interesting in a casual use of nilearn, you will not need
-numerics in Python. To go further, here are a few pointers.
+In case you plan to become a casual nileanr user, note that you will not need
```
Comment (lesteve) [inline doc/introduction.rst]: typo in nilearn

File: doc/introduction.rst (hunk 2 above)
Comment (AlexandreAbraham) [inline doc/introduction.rst]: :+1:

File: doc/introduction.rst (hunk 4 above)
Comment (AlexandreAbraham) [inline doc/introduction.rst]: Anaconda describes itself as "Anaconda Scientific Python Distribution".

File: doc/introduction.rst [hunk 13]
```diff
@@ -107,27 +123,29 @@ Glossary: machine learning vocabulary
...
 
 :Windows and MacOSX:
   We suggest you to install 64 bit Anaconda_.
 
   `Enthought Canopy`_, `PythonXY <http://code.google.com/p/pythonxy/>`_ are
-  other options. Enthought Canopy Express, the free version, should cover all
-  the required packages.
+  alternative options.
+  Enthought Canopy Express (download the free version) should provide all
+  the entirety of required packages.
```
Comment (AlexandreAbraham) [inline doc/introduction.rst]: Not sure that I prefer your formulation on this one. But can't tell why, that's totally subjective.

File: doc/introduction.rst [hunk 14]
```diff
@@ -148,8 +166,9 @@ a command prompt::
 
 .. warning::
 
-   Note that this is a "shell" command, that you need to type in a
-   command prompt, and not a Python command.
+   Note that this is a "shell" command, not a Python command. Concretely,
+   Mac users need to type this in the "Terminal", while Windows users
```
Comment (AlexandreAbraham) [inline doc/introduction.rst]: What about linux users?

File: doc/introduction.rst [hunk 15]
```diff
@@ -148,8 +166,9 @@ a command prompt::
...
+   need to type it in command line of cmd.exe.
```
Comment (AlexandreAbraham) [inline doc/introduction.rst]: `cmd.exe` send me back 6 or 7 years ago. We can cite it but I think that powershell should be evoked first.

File: doc/introduction.rst (hunk 5 above)
Comment (AlexandreAbraham) [inline doc/introduction.rst]: :+1: for "type in".

I would use "check" rather than "verify".

File: doc/introduction.rst [hunk 16]
```diff
@@ -158,42 +177,45 @@ a command prompt::
...
-If you do not get any errors, you have installed nilearn correctly.
+If no errors occur, you have installed nilearn correctly.
 
 Installing the development version
 ....................................
 
-**Downloading** As an alternative to using pip, and only in the case if you
-want the latest nilearn version you can do so by using git.
+**Downloading** If you really want the latest nilearn version, you can use
+git as an alternative to using pip.
```
Comment (AlexandreAbraham) [inline doc/introduction.rst]: "as an alternative to using pip" -> Should we put the line stating how to install the dev version with pip too?

File: doc/introduction.rst [hunk 17]
```diff
@@ -158,42 +177,45 @@ a command prompt::
...
 * **Under Linux**, run the following command (as a shell command, not a
   Python command)::
 
     git clone https://github.com/nilearn/nilearn.git
 
-As time goes, you can update your copy of nilearn by doing "git pull" in
-this directory.
+In the future, you can readily update your copy of nilearn
+by executing "git pull" in
+the nilearn root directory (yes, as a shell command).
```
Comment (AlexandreAbraham) [inline doc/introduction.rst]: I would remove the "yes, "

File: doc/introduction.rst [hunk 18]
```diff
@@ -201,20 +223,25 @@ same steps as in :ref:`testing_installation`.
...
-For a full blown introduction to
+For a full-blown introduction to
 using Python for science, see the `scipy lecture notes
 <http://scipy-lectures.github.io/>`_.
 
 
 We will be using `IPython <http://ipython.org>`_, which provides an
-interactive scientific environment. Start it with::
+interactive scientific environment that facilitates (e.g., measuring
+execution time of a command) and improves (e.g., printing of large matrices)
```
Comment (AlexandreAbraham) [inline doc/introduction.rst]: I don't think that measuring the execution time of a command is a useful feature for our users. I think that interactive debugging is more useful.

File: doc/introduction.rst [hunk 19]
```diff
@@ -234,27 +261,29 @@ which will open an interactive prompt::
...
 
 
 Your first steps with nilearn
 ------------------------------
 
-nilearn does not have a graphical user interface, it is used via Python
-code. Most important function to process neuroimaging data take as an
-input either a filename, or a `NiftiImage object
-<http://nipy.org/nibabel/nibabel_images.html>`_, which we call a
+First things first, nilearn does not have a graphical user interface.
```
Comment (AlexandreAbraham) [inline doc/introduction.rst]: What about "First things first, nilearn does not have a graphical user interface. But you will soon realize that you don't really need one."?

File: doc/introduction.rst (hunk 11 above)
Comment (AlexandreAbraham) [inline doc/introduction.rst]: Yes, I would stick to "with".

File: doc/introduction.rst [hunk 20]
```diff
@@ -309,16 +338,16 @@ To do a loop over each volume of a 4D image, use :func:`image.iter_img`::
       * Look at the "reference" section of the documentation: there is a
         function to compute the mean of a 4D image
 
-      * You can do a for loop in Python. You can use the "range" function
+      * To perform a for loop in Python, you can use the "range" function
 
-      * The solution is found :ref:`here
+      * The solution can be found at :ref:`here
```
Comment (AlexandreAbraham) [inline doc/introduction.rst]: "found at here" does not seem correct to my ear.

**Comment by AlexandreAbraham** [comment]
Great stuff. I just put in some subjective remarks but that may not be relevant since your english is better than mine.

**Comment by AlexandreAbraham** [comment]
@banilo do you plan to address the comments or should we close / take over this PR?

File: doc/introduction.rst (hunk 4 above)
Comment (banilo) [inline doc/introduction.rst]: year...I also prefer the new version

File: doc/introduction.rst (hunk 14 above)
Comment (banilo) [inline doc/introduction.rst]: i think, they just know

File: doc/introduction.rst (hunk 17 above)
Comment (banilo) [inline doc/introduction.rst]: that was for fun

File: doc/introduction.rst (hunk 5 above)
Comment (banilo) [inline doc/introduction.rst]: addressed all of these 3 but does not show here

**Comment by banilo** [comment]
> @banilo do you plan to address the comments or should we close / take over this PR?

sorry for the delay. I got sucked into something.

I should have addressed all immediately addressable suggestions now.

**Comment by lesteve** [comment]
Looks like a definitive improvement, thanks a lot for this.

**Comment by banilo** [comment]
Thanks for merging... I will look at the rest of the docs over the next days.

**Comment by lesteve** [comment]
Rather than improving the wording, I feel it is slightly higher priority to go over the documentation code snippets and make sure: 
* they make sense: the way we include part of the examples in the doc via litteralinclude is notoriously fragile
* they work. I am pretty sure some are broken for example https://github.com/nilearn/nilearn/issues/424 list a few of them

**Pull request merged**